core/src/test/scala/com/nvidia/spark/rapids/tool/profiling/AutoTunerSuite.scala

```diff
@@ -1507,4 +1578,135 @@ class AutoTunerSuite extends FunSuite with BeforeAndAfterEach with Logging {
+      inputSize: Double,
+      shuffleRead: Double,
+      gpuDevice: String,
+      expectedLines: Seq[String]): Unit = {
+    val customProps = mutable.LinkedHashMap(
+      "spark.executor.cores" -> "8",
+      "spark.executor.memory" -> "47222m",
+      "spark.rapids.sql.concurrentGpuTasks" -> "2",
+      "spark.task.resource.gpu.amount" -> "0.0625")
+    // mock the properties loaded from eventLog
+    val logEventsProps: mutable.Map[String, String] =
+      mutable.LinkedHashMap[String, String](
+        "spark.executor.cores" -> "16",
+        "spark.executor.instances" -> "1",
+        "spark.executor.memory" -> "80g",
+        "spark.executor.resource.gpu.amount" -> "1",
+        "spark.executor.instances" -> "1",
+        "spark.sql.shuffle.partitions" -> "200",
+        "spark.sql.files.maxPartitionBytes" -> "1g",
+        "spark.task.resource.gpu.amount" -> "0.0625",
+        "spark.rapids.memory.pinnedPool.size" -> "5g",
+        "spark.rapids.sql.enabled" -> "true",
+        "spark.plugins" -> "com.nvidia.spark.SQLPlugin",
+        "spark.rapids.sql.concurrentGpuTasks" -> "4")
+    val dataprocWorkerInfo = buildWorkerInfoAsString(Some(customProps), Some(32),
+      Some("212992MiB"), Some(5), Some(4), Some("15109MiB"), Some(gpuDevice))
+    val infoProvider = getMockInfoProvider(8126464.0, Seq(0), Seq(0.004), logEventsProps,
+      Some(defaultSparkVersion), meanInput = inputSize, meanShuffleRead = shuffleRead)
+    val autoTuner: AutoTuner = AutoTuner.buildAutoTunerFromProps(dataprocWorkerInfo, infoProvider)
+    val (properties, comments) = autoTuner.getRecommendedProperties()
+    val autoTunerOutput = Profiler.getAutoTunerResultsAsString(properties, comments)
+    assert(expectedLines.forall(line => autoTunerOutput.contains(line)),
+      s"Expected lines not found in AutoTuner output")
+  }
+
+  val testCases: TableFor4[Int, Int, String, Seq[String]] = Table(
+    ("inputSize", "shuffleRead", "gpuDevice", "expectedLines"),
+    // small input, small shuffle read
+    (1000, 1000, "T4", Seq(
+      "--conf spark.sql.adaptive.advisoryPartitionSizeInBytes=128m")),
+    // large input, small shuffle read
+    (40000, 1000, "T4", Seq()),
+    // large input, large shuffle read
+    (40000, 80000, "T4", Seq(
+      "--conf spark.sql.adaptive.advisoryPartitionSizeInBytes=32m",
+      "--conf spark.sql.adaptive.coalescePartitions.initialPartitionNum=800",
+      "--conf spark.sql.adaptive.coalescePartitions.parallelismFirst=false"
+    )),
+    // large input, large shuffle read, faster GPU
+    (40000, 80000, "A100", Seq(
+      "--conf spark.sql.adaptive.advisoryPartitionSizeInBytes=64m",
+      "--conf spark.sql.adaptive.coalescePartitions.initialPartitionNum=400",
+      "--conf spark.sql.adaptive.coalescePartitions.parallelismFirst=false"
+    ))
+  )
+
+  forAll(testCases) { (inputSize, shuffleRead, gpuDevice, expectedLines) =>
+    test(s"AutoTuner test for AQE partition configs - input size: $inputSize," +
```

Comment:
The autotuner test is redundant as we know it is a test in the class AutoTunerSuite
```suggestion
    test(s"AQE partition configs - input size: $inputSize," +
```